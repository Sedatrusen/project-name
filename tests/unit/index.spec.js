import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import index from '@/pages/index.vue'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('index sayfası', () => {
  let actions
  let store
  let getters
  let state
  beforeEach(() => {
    state = { products: { Products: [{ id: '1' }, { id: '2' }] } }
    actions = {
      addproducts: jest.fn(),
    }

    store = new Vuex.Store({
      actions,
      getters,
      state,
    })
  })

  it('index sayfası çalışıyor', () => {
    const wrapper = shallowMount(index, {
      localVue,
      store,
    })
    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html).toMatchSnapshot()
  })
  it('yazı gözüküyor', () => {
    const wrapper = shallowMount(index, { localVue, store })
    const p = wrapper.find('h1')
    expect(p.text()).toBe('Recommended')
  })
})
