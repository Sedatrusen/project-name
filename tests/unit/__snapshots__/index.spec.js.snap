// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index sayfası index sayfası çalışıyor 1`] = `[Function]`;
